pkg/services/publicdashboards/service/service.go

```diff
@@ -257,11 +261,20 @@ func (pd *PublicDashboardServiceImpl) buildMetricRequest(ctx context.Context, da
 
 	// determine safe resolution to query data at
 	safeInterval, safeResolution := pd.getSafeIntervalAndMaxDataPoints(reqDTO, ts)
-	for i := range queries {
-		queries[i].Set("intervalMs", safeInterval)
-		queries[i].Set("maxDataPoints", safeResolution)
+
+	n := 0
+	for _, q := range queries {
+		mapQuery := q.MustMap()
+		if mapQuery["hide"] != true {
+			queries[n] = q
+			queries[n].Set("intervalMs", safeInterval)
+			queries[n].Set("maxDataPoints", safeResolution)
+			n++
+		}
```

Comment:
I would put this logic inside `GroupQueriesByPanelId` where we decide which queries needs to get executed.